src/components/backup/RestoreCloudStep.tsx

```diff
@@ -76,33 +86,40 @@ const KeyboardSizeView = styled(KeyboardArea)({
 
 type RestoreCloudStepParams = {
   RestoreSheet: {
-    selectedBackup: Backup;
+    selectedBackup: BackupFile;
   };
 };
 
 export default function RestoreCloudStep() {
   const { params } = useRoute<RouteProp<RestoreCloudStepParams & RestoreSheetParams, 'RestoreSheet'>>();
-
-  const { userData } = useCloudBackups();
+  const { password } = backupsStore(state => ({
+    password: state.password,
+  }));
 
   const { selectedBackup } = params;
   const { isDarkMode } = useTheme();
-  const [loading, setLoading] = useState(false);
+  const { canGoBack, goBack } = useNavigation();
+
+  const onRestoreSuccess = useCallback(() => {
+    while (canGoBack()) {
+      goBack();
+    }
```

Comment:
isn't the `reset` useful here? https://reactnavigation.org/docs/6.x/navigation-actions/#reset